lib/srv/sess.go

```diff
@@ -424,12 +433,44 @@ type sessionRecorder struct {
 	sid rsession.ID
 }
 
+func newSessionRecorder(alog events.IAuditLog, sid rsession.ID) *sessionRecorder {
+	sr := &sessionRecorder{
+		alog: alog,
+		sid:  sid,
+	}
+	return sr
+}
+
 // Write takes a chunk and writes it into the audit log
 func (r *sessionRecorder) Write(data []byte) (int, error) {
+	const (
+		minChunkLen   = 4
+		maxDelay      = time.Millisecond * 20
+		requiredDelay = time.Millisecond * 5
+	)
+	// terminal recording is a tricky business. the TTY subsystem expects a certain
+	// delay when it writes to a virtual console. In our case recording takes no time
+	// so we have to emulate a bit of a delay here:
+	var start time.Time
+	dataLen := len(data)
+	if dataLen > minChunkLen {
```

Comment:
added more comments. the idea is to `sleep(t)` where `t=min(chunk_size, max_delay)`